### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by ahartikainen**
Update travis to use matrix structure.

Upgrade distribution to xenial.

Fix pyro code.

~Fix error in plotting code which caused tests to fail.~ Pyro fix was the answer.

### CODE REVIEW

**Comment by ahartikainen** [comment]
This is ready now. Only think I don't like is the squeezing pyro output.

Maybe that could be done bu squeezing out only the last dimension if the shape is (N,1)?

File: arviz/plots/densityplot.py [hunk 1]
```diff
@@ -171,10 +171,9 @@ def plot_density(
     for plotters in to_plot:
         length_plotters.append(len(plotters))
         for var_name, selection, _ in plotters:
-            label = make_label(var_name, selection)
+            label = make_label(var_name, selection).replace("\n", "_")
```
Comment (canyon289) [inline arviz/plots/densityplot.py]: I think it's better if `.replace` be done in `make_label`. perhaps by passing in a tuple. That way all the label making logic is in one place and `make_label` provides the finished label. 

Thoughts?

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
 
-python:
-  - "3.6"
-  - "3.5"
+language: python
 
-env:
-  - PYSTAN_VERSION=2.17
-  - PYSTAN_VERSION=latest
+matrix:
```
Comment (canyon289) [inline .travis.yml]: This is matrix approach is way better. Thanks for doing this

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+  include:
+  - name: "Python 3.6 Unit Test"
+    python: 3.6
+    env: PYSTAN_VERSION=latest NAME="UNIT"
+  - name: "Python 3.5 Unit Test"
+    python: 3.5
+    env: PYSTAN_VERSION=latest NAME="UNIT"
+  - name: "Python 3.6 Lint"
+    python: 3.6
+    env: PYSTAN_VERSION=latest NAME="LINT"
```
Comment (canyon289) [inline .travis.yml]: No change required but why have a separate lint environment?

File: arviz/plots/densityplot.py (hunk 1 above)
Comment (canyon289) [reply arviz/plots/densityplot.py]: Will fix on another PR per discussion. Thanks ari :)

**Comment by ahartikainen** [comment]
If tests pass, then this is good to go.

We will need to think about pyro shape issue later and also update densityplot if multiple models contain different parameters.

    {'a', 'b'} + {'b','c'} --> should make 3 subplots, now it makes 2.

**Pull request merged**